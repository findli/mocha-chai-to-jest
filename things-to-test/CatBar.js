import React from 'react';
import StatelessAnimal from './StatelessAnimal';
import HatchingAnimal from './HatchingAnimal';

const CatBar = () => (
    <div className="catbar">
        <p>
            I am a cat bar. I contain cats.
        </p>
        <StatelessAnimal sound="Meow!" />
        <StatelessAnimal sound="Meoooow!" />
        <StatelessAnimal sound="Meeeeeeeeow!" />
        <HatchingAnimal />
    </div>
);

CatBar.displayName = 'CatBar';

export default CatBar;
