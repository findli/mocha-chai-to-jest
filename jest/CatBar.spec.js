import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import CatBar from '../things-to-test/CatBar';

describe('CatBar component', () => {
    it('renders correctly with min params', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <CatBar />,
        );
        expect(result).toMatchSnapshot();
    });
});
