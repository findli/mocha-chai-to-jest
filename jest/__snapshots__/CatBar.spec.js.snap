// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatBar component renders correctly with min params 1`] = `
<div
  className="catbar"
>
  <p>
    I am a cat bar. I contain cats.
  </p>
  <StatelessAnimal
    sound="Meow!"
  />
  <StatelessAnimal
    sound="Meoooow!"
  />
  <StatelessAnimal
    sound="Meeeeeeeeow!"
  />
  <HatchingAnimal />
</div>
`;
